inference-engine/src/legacy_api/src/ie_cnn_layer_builder_ngraph.cpp

```diff
@@ -1682,8 +1682,37 @@ CNNLayer::Ptr NodeConverter<ngraph::op::TopKIE>::createLayer(const std::shared_p
     auto castedLayer = ngraph::as_type_ptr<ngraph::op::TopKIE>(layer);
     if (castedLayer == nullptr) THROW_IE_EXCEPTION << "Cannot get " << params.type << " layer " << params.name;
 
-    res->params["mode"] = castedLayer->get_mode();
-    res->params["sort"] = castedLayer->get_sort_type();
+    auto mode = castedLayer->get_mode();
+    std::string str_mode;
+    switch (mode) {
+    case ngraph::op::v1::TopK::Mode::MIN:
+        str_mode = "min";
+        break;
+    case ngraph::op::v1::TopK::Mode::MAX:
+        str_mode = "max";
+        break;
+    default:
+        THROW_IE_EXCEPTION << "Unsupported TopK mode";
+    }
+
+    auto sort = castedLayer->get_sort_type();
+    std::string str_sort;
+    switch (sort) {
```

Comment:
Doesn't `as_string<enum_type>(sort)` do the work?

Reply:
Done.